src/xbot/common/logging/PeriodicDeltaObserver.java

```diff
@@ -0,0 +1,36 @@
+package xbot.common.logging;
+
+import org.apache.log4j.Logger;
+
+public class PeriodicDeltaObserver {
+
+    protected static Logger log = Logger.getLogger(PeriodicDeltaObserver.class);
+    private final String callerName;
+    private String message;
+    
+    private double checkPowerThreshold;
+    
+    private double oldPower = 0;
+    
+    public PeriodicDeltaObserver(String callerName, String message, double checkPowerThreshold) {
+        this.callerName = callerName;
+        this.message = message;
+        this.checkPowerThreshold = checkPowerThreshold;
+    }
+    
+    public void setMessage(String message) {
```

Comment:
I think you should change this so that the class itself only supplies a Boolean which indicates whether the delta was passed, and the consumer does the logging.